src/metric/definition.go

```diff
@@ -1017,11 +1027,46 @@ func isPersistentVolume() definition.FetchFunc {
 	}
 }
 
-func computePercentage(current, all uint64) (definition.FetchedValue, error) {
-	if all == uint64(0) {
-		return nil, errors.New("division by zero")
+func computePercentage(dividend, divisor interface{}) (definition.FetchedValue, error) {
+	var a, b float64
+
+	a, err := convertValue(dividend)
+	if err != nil {
+		return nil, fmt.Errorf("casting dividend: %w", err)
+	}
+
+	b, err = convertValue(divisor)
+	if err != nil {
+		return nil, fmt.Errorf("casting divisor: %w", err)
+	}
+
+	if b == float64(0) {
+		return nil, fmt.Errorf("division by zero")
+	}
+
+	return a / b * 100, nil
+}
+
+func convertValue(v interface{}) (float64, error) {
+	switch v := v.(type) {
+	case uint64:
+		return float64(v), nil
+	case int:
+		return float64(v), nil
+	case prometheus.GaugeValue:
+		return float64(v), nil
+	case float64:
+		return v, nil
```

Comment:
I'd say we can shorten the switch statement like this:
```suggestion
	case uint64, int, float64, prometheus.GaugeValue:
		return float64(v), nil
```

Reply:
![Screen Shot 2022-01-07 at 12 42 29 PM](https://user-images.githubusercontent.com/43335750/148539258-8f20416d-c4fa-4570-8cd4-5999ce90fde9.png)
That was my first implementation, but sadly go compiler is not smart enough. I found it really weird as well that was not possible

Reply:
fuck, you're totally right.